opentelemetry-sdk/src/opentelemetry/sdk/_metrics/instrument.py

```diff
@@ -51,7 +51,15 @@ def __init__(
         unit: str = "",
         description: str = "",
     ):
-        self.name = name
+        # pylint: disable=no-member
+        is_name_valid, is_unit_valid = self._check_name_and_unit(name, unit)
+
+        if not is_name_valid:
+            raise Exception(f"Invalid name received {name}")
```

Comment:
This message can be made little more useful to something like "Expected ASCII string of max length 63 but got "?